ietf-network-inventory.yang

```diff
@@ -96,10 +96,103 @@ module ietf-network-inventory {
         "A physical network element (NE). ";
     }
 
+  /*
+   * Editors' Note: This identity may need to be moved to
+   *                iana-hardware update
+   */
+
+  identity transceivers-module {
+    base ianahw:hardware-class;
+    description
+      "This identity is applicable if the hardware class is some sort
+       of self-contained sub-system which contains one or more
+       transceivers (e.g., optical or electrical transceivers) which
+       are transmitting and receiving a signal from the media.
+       
+       A transceivers-module component can only be contained by a
+       port component.";
+  }
+
+  /*
+   * Types
+   */
+
+
+  typedef ne-ref {
+    type leafref {
+      path  "/nwi:network-inventory/nwi:network-elements"
+          + "/nwi:network-element/nwi:ne-id";
+    }
+    description
+      "This type is intended to be used by data models that need to
+      reference Network Element.";
+  }
+
   /*
    * Groupings
    */
 
+  grouping port-ref {
+    description
+      "This grouping is intended to be used by data models that need
+      to reference a port component within a Network Element.";
+    leaf ne-ref {
+      type nwi:ne-ref;
+      description
+        "The reference to the Network Element which contains the
+        port to be referenced.";
+    }
+    leaf port-ref {
+      type leafref {
+        path  "/nwi:network-inventory/nwi:network-elements"
+            + "/nwi:network-element[nwi:ne-id=current()/../ne-ref]/"
+            + "nwi:components/nwi:component/nwi:component-id";
+      }
+      must  "derived-from-or-self (/nwi:network-inventory/"
+          + "nwi:network-elements/nwi:network-element[nwi:ne-id="
+          + "current()/../ne-ref]/nwi:components/"
+          + "nwi:component[nwi:component-id=current()]/nwi:/class,"
+          + "'ianahw:port')";
+      description
+        "The reference to the port component.";
+    }
+  }
+
+  grouping channel-ref {
+    description
+      "This grouping is intended to be used by data models that need
+      to reference one or more channels within a transceivers module
+      component.";
+    leaf ne-ref {
+      type nwi:ne-ref;
+      description
+        "The reference to the Network Element which contains the
+        port to be referenced.";
+    }
+    leaf transceivers-module-ref {
+      type leafref {
+        path  "/nwi:network-inventory/nwi:network-elements"
+            + "/nwi:network-element[nwi:ne-id=current()/../ne-ref]/"
+            + "nwi:components/nwi:component/nwi:component-id";
+      }
+      description
+        "The reference to the transceivers module component.";
+    }
+    leaf-list channel-number {
```

Comment:
What these channels are representing with respect OpenConfig ? physical-channels that are related to different logical-interface in which the port is broken ?